src/output/dnssim/tls.c

```diff
@@ -104,7 +104,7 @@ void _output_dnssim_tls_process_input_data(_output_dnssim_connection_t* conn)
     /* See https://gnutls.org/manual/html_node/Data-transfer-and-termination.html#Data-transfer-and-termination */
     while (true) {
         /* Connection might have been closed due to an error, don't try to use it. */
-        if (conn->state != _OUTPUT_DNSSIM_CONN_ACTIVE)
+        if (conn->state < _OUTPUT_DNSSIM_CONN_ACTIVE || conn->state >= _OUTPUT_DNSSIM_CONN_CLOSING)
```

Comment:
This smells bad to me, if you move around the numbering of the states this will break. Why not `conn->state != _OUTPUT_DNSSIM_CONN_ACTIVE && conn->state != _OUTPUT_DNSSIM_CONN_CONGESTED` ?

Reply:
The states are intentionally numbered in the specific order to allow range-checks like this. I think it's more likely to add a new state that this range should (and will) cover, rather than renumbering or changing order of the states.

I already use a similar range here: https://github.com/DNS-OARC/dnsjit/blob/develop/src/output/dnssim/connection.c#L34

Reply:
There is no documentation that the states are used liked this. I would recommend either change the way you check the states to a `switch` or document it clearly in the state definitions and where they are used.

Reply:
I've added a comment to the enum to document this intention.